site/app/models/DisplayImage.php

```diff
@@ -0,0 +1,212 @@
+<?php
+
+namespace app\models;
+
+use app\exceptions\BadArgumentException;
+use app\exceptions\FileReadException;
+use app\exceptions\FileWriteException;
+use app\libraries\Core;
+use app\libraries\FileUtils;
+
+/**
+ * Class DisplayImage
+ * @package app\models\image
+ *
+ * DisplayImage model represents a user's personal head shot.  Users may have multiple images in their user_data
+ * directory, and DisplayImage takes care of selecting the single most appropriate image for display.
+ *
+ * @method string getDisplayImageState()
+ * @method string setDisplayImageState()
+ * @method string getPath()
+ * @method string setPath()
+ * @method string getMimeType()
+ * @method string setMimeType()
+ */
+class DisplayImage extends AbstractModel {
+    /** string[] The set of legal display_image_state which may be collected from the DB */
+    const LEGAL_IMAGE_STATES = ['system', 'preferred', 'flagged'];
+
+    /** string[] The set of directories for which it is legal to save a DisplayImage to */
+    const LEGAL_FOLDERS = ['system_images', 'user_images'];
+
+    /**
+     * int
+     * When a new DisplayImage is saved, this is the maximum dimension either side may be.  Non-square images will
+     * have the larger dimension resized to this constant, while the other dimension will be reduced to maintain the
+     * original form factor of the image.
+     */
+    const IMG_MAX_DIMENSION = 500;
+
+    /** @prop @var string Display image state, should match database field */
+    protected $display_image_state;
+
+    /** @prop @var string The file path to the selected image */
+    protected $path;
+
+    /** @prop @var string The image mime type */
+    protected $mime_type;
+
+    /**
+     * DisplayImage constructor
+     *
+     * Determines the path to the display image based on the display image state.  Also tests to ensure the image
+     * is readable at instantiation time.
+     *
+     * @param Core $core
+     * @param string $user_id
+     * @param string $display_image_state
+     * @throws BadArgumentException An illegal display_image_state was passed in
+     * @throws FileReadException Unable to read the file at the selected location for whatever reason
+     * @throws \ImagickException
+     */
+    public function __construct(Core $core, string $user_id, string $display_image_state) {
+        parent::__construct($core);
+
+        // Validate passed image state
+        if (!in_array($display_image_state, self::LEGAL_IMAGE_STATES)) {
+            throw new BadArgumentException('Unknown display_image_state!');
+        }
+
+        $sub_dir = null;
+        if ($display_image_state === 'system' || $display_image_state === 'flagged') {
+            $sub_dir = 'system_images';
+        }
+        elseif ($display_image_state === 'preferred') {
+            $sub_dir = 'user_images';
+        }
+
```

Comment:
There are only 3 image states right - can this be re-written as 
```php
sub_dir = $display_image_sate === 'preferred' ? 'user_images' : 'system_images';
```

Reply:
Changed